### PR TITLE
ProxyTrace should check for tests locations

### LINKED ISSUE #1480: Error: Expected a value other than null.
I get this error too. I'm a brand new user and have infection/infection 0.21.0@dfacb1e3d9def7fd34c6fb8df4d4bd098815dc82 PHPUnit version 8.5.14 PHP 7.4.15 Xdebug 3.0.2 codebase is CakePHP 4.2.2. 

```
challgren@challgren-pro authenticon-web %  XDEBUG_MODE=coverage; vendor/bin/infection

    ____      ____          __  _
   /  _/___  / __/__  _____/ /_(_)___  ____
   / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
 _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
/___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/

Infection - PHP Mutation Testing Framework version 0.21.0@dfacb1e3d9def7fd34c6fb8df4d4bd098815dc82


Running initial test suite...

PHPUnit version: 8.5.14

  173 [============================] 19 secs

Generate mutants...

Processing source code files: 33/36
In Assert.php line 2042:
                                     
  Expected a value other than null.  
                                     

run [--test-framework TEST-FRAMEWORK] [--test-framework-options TEST-FRAMEWORK-OPTIONS] [-j|--threads THREADS] [--only-covered] [-s|--show-mutations] [--no-progress] [--force-progress] [-c|--configuration CONFIGURATION] [--coverage COVERAGE] [--mutators MUTATORS] [--filter FILTER] [--formatter FORMATTER] [--git-diff-filter GIT-DIFF-FILTER] [--git-diff-base GIT-DIFF-BASE] [--logger-github] [--noop] [--min-msi MIN-MSI] [--min-covered-msi MIN-COVERED-MSI] [--log-verbosity LOG-VERBOSITY] [--initial-tests-php-options INITIAL-TESTS-PHP-OPTIONS] [--skip-initial-tests] [--ignore-msi-with-no-mutations] [--debug] [--dry-run]


```

See details https://github.com/infection/infection/issues/1357#issuecomment-777754308

_Originally posted by @challgren in https://github.com/infection/infection/issues/1357#issuecomment-777741442_

### PR DESCRIPTION
**Pull request by sanmai**
Fixes #1480

This PR:

- [x] Fixes #1480
- [x] Targets 0.21
- [x] Covered by tests

### CODE REVIEW

File: src/TestFramework/Coverage/ProxyTrace.php [hunk 1]
```diff
@@ -108,6 +108,10 @@ public function getTests(): ?TestLocations
 
     public function getAllTestsForMutation(NodeLineRangeData $lineRange, bool $isOnFunctionSignature): iterable
     {
+        if ($this->lazyTestLocations === null) {
+            return [];
+        }
```
Comment (sanmai) [inline src/TestFramework/Coverage/ProxyTrace.php:113]: `NodeMutationGenerator` does not check if there are tests, just asks for them. It is not their call to know the details, so we're doing it here.

**Comment by sanmai** [comment]
Ow, how did it automerge without a review?

**Comment by sanmai** [comment]
I guess we're all set then!

**Comment by maks-rafalko** [comment]
> Ow, how did it automerge without a review?

this is because the target is `0.21`, not `master`, and we didn't have "Require approves to merge" setting for `0.*` branches. Fixed.

Thank you @sanmai

**Pull request merged**